src/core/value.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const GlobalBigNumber = require('bignumber.js');
+
+const BigNumber = GlobalBigNumber.another({
+  ROUNDING_MODE: GlobalBigNumber.ROUND_HALF_UP,
+  DECIMAL_PLACES: 40
+});
+
+function Value(value) {
+  this._value = new BigNumber(value);
+}
+
+Value.prototype.abs = function() {
+  let result = this._value.abs();
+  return this._canonicalize(result);
+};
+
+Value.prototype.add = function(addend) {
+  let result = this._value.plus(addend._value);
```

Comment:
Should probably `assert(addend instanceof Value)` or `if (!(addend instanceof Value)) myAddend = new Value(addend)`. I also think we should add accessors instead of ._value, but not important